ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml

```diff
@@ -299,20 +299,35 @@ objects:
         echo "Removing bootstrap host from control plane api pool"
         (cd /tmp/artifacts/terraform && terraform apply -auto-approve=true -var=bootstrap_dns="false")
 
-        echo "Approving any pending CSR requests"
-        oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
+        function approve_csrs() {
+          while true; do
+            if [[ ! -f /tmp/install-complete ]]; then
+              oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs --no-run-if-empty oc adm certificate approve
+              sleep 15 & wait
+              continue
+            else
+              break
+            fi
+          done
+        }
 
-        until oc get configs.imageregistry.operator.openshift.io cluster >/dev/null
-        do
-          echo "waiting for configs.imageregistry.operator.openshift.io cluster to exit failed. Retrying in 1 minute..."
-          sleep 1m
-        done
+        function update_image_registry() {
+          while true; do
+            sleep 10;
+            oc get configs.imageregistry.operator.openshift.io/cluster > /dev/null && break
+          done
+          oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+        }
+
+        echo "Approving pending CSRs"
+        approve_csrs &
 
-        oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+        update_image_registry &
 
         echo "Completing UPI setup"
         openshift-install --dir=/tmp/artifacts/installer wait-for install-complete &
         wait "$!"
+        touch /tmp/install-complete
```

Comment:
Do we need this?  We already have [this `trap`][1] to set either `/tmp/setup-success` or `/tmp/exit`.

[1]: https://github.com/openshift/release/blob/d143934d7408a140bcb4037c29c9cf0fadfb293d/ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml#L220

Reply:
its from https://github.com/openshift/release/pull/3573 .
need the approve_csr to keep looping until install-complete. Any other suggestions?

Reply:
> need the approve_csr to keep looping until install-complete.

Guard for both (like [this][1])?

[1]: https://github.com/openshift/release/blob/d143934d7408a140bcb4037c29c9cf0fadfb293d/ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml#L129-L136

Reply:
Also, this script is [`set -e`][1], so `touch /tmp/install-complete` ends up with the same semantics as the existing `/tmp/setup-success`, not the success-agnostic completion marker that #3473 seems to have expected.

[1]: https://github.com/openshift/release/blob/d143934d7408a140bcb4037c29c9cf0fadfb293d/ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml#L118

Reply:
WDYT https://github.com/openshift/release/compare/0217808359514ff9f0bd29772bc558451ccbb236...dd9f801d96c4bf4b42ecae1e04f7a6e5915e8b36